HPX build fails with std::atomic missing error

## Expected Behavior

building hpx from source

## Actual Behavior

[grostig@fx8150-grostig-com build]$ cmake -DCMAKE_INSTALL_PREFIX=/home/grostig/libs ..
-- CMake version: 3.12.1
-- HPX version: 1.3.0
-- No vectorization library configured
-- Native TLS is enabled.
-- Nice threadlevel is disabled.
-- Stack traces are enabled.
-- Compiling with the native toolset
-- C++ mode enforced: C++17
-- Performing Test HPX_WITH_CXX11_ALIAS_TEMPLATES - pre-set to TRUE
-- Performing Test HPX_WITH_CXX11_ALIGNAS - pre-set to TRUE
-- Performing Test HPX_WITH_CXX11_AUTO - pre-set to TRUE
-- Performing Test HPX_WITH_CXX11_CONSTEXPR - pre-set to TRUE
-- Performing Test HPX_WITH_CXX11_DECLTYPE - pre-set to TRUE
-- Performing Test HPX_WITH_CXX11_SFINAE_EXPRESSION - pre-set to TRUE
-- Performing Test HPX_WITH_CXX11_DEFAULTED_FUNCTIONS - pre-set to TRUE
-- Performing Test HPX_WITH_CXX11_DELETED_FUNCTIONS - pre-set to TRUE
-- Performing Test HPX_WITH_CXX11_EXPLICIT_CONVERSION_OPERATORS - pre-set to TRUE
-- Performing Test HPX_WITH_CXX11_EXPLICIT_VARIADIC_TEMPLATES - pre-set to TRUE
-- Performing Test HPX_WITH_CXX11_EXTENDED_FRIEND_DECLARATIONS - pre-set to TRUE
-- Performing Test HPX_WITH_CXX11_FUNCTION_TEMPLATE_DEFAULT_ARGS - pre-set to TRUE
-- Performing Test HPX_WITH_CXX11_INLINE_NAMESPACES - pre-set to TRUE
-- Performing Test HPX_WITH_CXX11_LAMBDAS - pre-set to TRUE
-- Performing Test HPX_WITH_CXX11_NOEXCEPT - pre-set to TRUE
-- Performing Test HPX_WITH_CXX11_NULLPTR - pre-set to TRUE
-- Performing Test HPX_WITH_CXX11_NSDMI - pre-set to TRUE
-- Performing Test HPX_WITH_CXX11_RANGE_BASED_FOR - pre-set to TRUE
-- Performing Test HPX_WITH_CXX11_RVALUE_REFERENCES - pre-set to TRUE
-- Performing Test HPX_WITH_CXX11_SCOPED_ENUMS - pre-set to TRUE
-- Performing Test HPX_WITH_CXX11_STATIC_ASSERT - pre-set to TRUE
-- Performing Test HPX_WITH_CXX11_VARIADIC_MACROS - pre-set to TRUE
-- Performing Test HPX_WITH_CXX11_VARIADIC_TEMPLATES - pre-set to TRUE
-- Performing Test HPX_WITH_CXX11_OVERRIDE - pre-set to TRUE
-- Performing Test HPX_WITH_CXX11_ARRAY - pre-set to TRUE
-- Performing Test HPX_WITH_CXX11_ATOMIC - Failed
-- Test failed, detailed output:


CMake Error at cmake/HPX_Message.cmake:47 (message):
  HPX needs support for C++11 std::atomic
Call Stack (most recent call first):
  cmake/HPX_AddConfigTest.cmake:137 (hpx_error)
  cmake/HPX_AddConfigTest.cmake:406 (add_hpx_config_test)
  cmake/HPX_PerformCxxFeatureTests.cmake:92 (hpx_check_for_cxx11_std_atomic)
  CMakeLists.txt:1174 (hpx_perform_cxx_feature_tests)


-- Configuring incomplete, errors occurred!
See also "/home/grostig/src/hpx/hpx_project_source/hpx/build/CMakeFiles/CMakeOutput.log".
See also "/home/grostig/src/hpx/hpx_project_source/hpx/build/CMakeFiles/CMakeError.log".


+++++++++++++++++++++++++++++++
This is the cmake error log:

Determining if the function __atomic_fetch_add_4 exists in the atomic failed with the following output:
Change Dir: /home/grostig/src/hpx/hpx_project_source/hpx/build/CMakeFiles/CMakeTmp

Run Build Command:"/usr/bin/gmake" "cmTC_b717f/fast"
/usr/bin/gmake -f CMakeFiles/cmTC_b717f.dir/build.make CMakeFiles/cmTC_b717f.dir/build
gmake[1]: Entering directory '/home/grostig/src/hpx/hpx_project_source/hpx/build/CMakeFiles/CMakeTmp'
Building C object CMakeFiles/cmTC_b717f.dir/CheckFunctionExists.c.o
/usr/lib64/ccache/cc   -DCHECK_FUNCTION_EXISTS=__atomic_fetch_add_4   -o CMakeFiles/cmTC_b717f.dir/CheckFunctionExists.c.o   -c /usr/share/cmake/Modules/CheckFunctionExists.c
<command-line>: warning: conflicting types for built-in function â__atomic_fetch_add_4â [-Wbuiltin-declaration-mismatch]
/usr/share/cmake/Modules/CheckFunctionExists.c:7:3: note: in expansion of macro âCHECK_FUNCTION_EXISTSâ
   CHECK_FUNCTION_EXISTS(void);
   ^~~~~~~~~~~~~~~~~~~~~
Linking C executable cmTC_b717f
/usr/bin/cmake -E cmake_link_script CMakeFiles/cmTC_b717f.dir/link.txt --verbose=1
/usr/lib64/ccache/cc  -DCHECK_FUNCTION_EXISTS=__atomic_fetch_add_4    -rdynamic CMakeFiles/cmTC_b717f.dir/CheckFunctionExists.c.o  -o cmTC_b717f -latomic 
/usr/bin/ld: cannot find /usr/lib64/libatomic.so.1.2.0 inside /
collect2: error: ld returned 1 exit status
gmake[1]: *** [CMakeFiles/cmTC_b717f.dir/build.make:87: cmTC_b717f] Error 1
gmake[1]: Leaving directory '/home/grostig/src/hpx/hpx_project_source/hpx/build/CMakeFiles/CMakeTmp'
gmake: *** [Makefile:121: cmTC_b717f/fast] Error 2


## Steps to Reproduce the Problem

cmake -DCMAKE_INSTALL_PREFIX=/home/grostig/libs ..

## Specifications

  - HPX Version: Master as of today.
  - Platform (compiler, OS): fedora29 with libatomic installed.

## My Note:

It seems to be looking for:
/usr/lib64/libatomic.so.1.2.0

My system is up to date and has this version installed:
ls -l libatomic*
lrwxrwxrwx. 1 root root    26 Aug  7  2018 libatomic_ops_gpl.so.1 -> libatomic_ops_gpl.so.1.1.2
-rwxr-xr-x. 1 root root 15392 Aug  7  2018 libatomic_ops_gpl.so.1.1.2
lrwxrwxrwx. 1 root root    22 Aug  7  2018 libatomic_ops.so.1 -> libatomic_ops.so.1.1.1
-rwxr-xr-x. 1 root root 14984 Aug  7  2018 libatomic_ops.so.1.1.1
lrwxrwxrwx. 1 root root    18 Dec 16 11:05 libatomic.so.1 -> libatomic.so.1.2.0
-rwxr-xr-x. 1 root root 31128 Dec 16 11:10 libatomic.so.1.2.0

--- a/cmake/HPX_AddConfigTest.cmake
+++ b/cmake/HPX_AddConfigTest.cmake
@@ -396,6 +396,11 @@ endfunction()
 
 ###############################################################################
 function(hpx_check_for_cxx11_std_atomic)
+  # Make sure HPX_HAVE_LIBATOMIC is removed from the cache if necessary
+  if(NOT HPX_WITH_CXX11_ATOMIC)
+    unset(HPX_HAVE_LIBATOMIC CACHE)
+  endif()
+
   # Sometimes linking against libatomic is required for atomic ops, if
   # the platform doesn't support lock-free atomics.
   check_library_exists(atomic __atomic_fetch_add_4 "" HPX_HAVE_LIBATOMIC)

